### DISCUSSION TITLE
Recommended Connection Settings For AWS Lambda?

### DISCUSSION THREAD

**Discussion by bestickley** [porsager/postgres#425] [Q&A] [answered]
Hi, love this package! Makes queries a breeze. I'm using this in AWS Lambda. Does anyone have any recommendations for optimal connecting settings that don't hold on to connections too long which aren't ideal for a "serverless" environment? Ideally, I'd like something like `sql.end()` like from [serverless-mysql] (https://github.com/jeremydaly/serverless-mysql) which does "connection management". 

![image](https://user-images.githubusercontent.com/35735118/178242029-c9f02564-95a2-4223-9dda-dc13e6efca8c.png)

Thank you!

**Comment by bestickley** [comment]
Based on [How To: Manage RDS Connections from AWS Serverless Lamda Functions](https://www.jeremydaly.com/manage-rds-connections-aws-lambda/) reducing the `max_timeout` to 10 minutes is maybe a start?

**Comment by porsager** [comment, marked as answer]
Thanks a lot 😊 

You'll properly know best based on your functions requirements, but setting idle_timeout to 0 will effectively close the connection as soon as no queries are queued.  If you do several queries with other work inbetween perhaps 30 seconds is a good value.
